apps/src/dance/ai/utils.ts

```diff
@@ -53,3 +53,24 @@ export const getLabelMap = (
   });
   return map;
 };
+
+/**
+ * Generate code that can be executed to preview the output of the AI-generated blocks.
+ */
+export const generatePreviewCode = (
+  workspace: Workspace,
+  resultJsonString: string
+): string => {
+  const blocks = generateBlocksFromResult(workspace, resultJsonString);
+  // Create a temporary setup block
+  const setup: BlockSvg = workspace.newBlock('Dancelab_whenSetup') as BlockSvg;
+
+  // Attach the blocks to the setup block
+  setup.getInput('DO')?.connection?.connect(blocks[0].previousConnection);
+
```

Comment:
Just curious why you don't have to explicitly remove the temp setup block from the workspace so it doesn't remain after preview.

Reply:
In the previous version, we were creating and these blocks in the main workspace (`Blockly.getMainWorkspace()`) but now since we're passing in a temporary workspace, we don't need to necessarily remove the block. The workspace gets disposed after using this function anyway (in `DanceAiModal`)